lib/class-convertkit-api.php

```diff
@@ -0,0 +1,1580 @@
+		$email      = trim( $email );
+		$first_name = trim( $first_name );
+
+		// Return error if no Form ID or email address is specified.
+		if ( empty( $form_id ) ) {
+			return new WP_Error( 'convertkit_api_error', $this->get_error_message( 'form_subscribe_form_id_empty' ) );
+		}
+		if ( empty( $email ) ) {
+			return new WP_Error( 'convertkit_api_error', $this->get_error_message( 'form_subscribe_email_empty' ) );
+		}
+
+		// Build request parameters.
+		$params = array(
+			'api_key'    => $this->api_key,
+			'email'      => $email,
+			'first_name' => $first_name,
+		);
+		if ( $fields ) {
+			$params['fields'] = $fields;
+		}
+		if ( $tag_ids ) {
+			$params['tags'] = $tag_ids;
+		}
+
+		// Send request.
+		$response = $this->post( 'forms/' . $form_id . '/subscribe', $params );
+
+		// If an error occured, log and return it now.
+		if ( is_wp_error( $response ) ) {
+			$this->log( 'API: form_subscribe(): Error: ' . $response->get_error_message() );
+			return $response;
+		}
+
+		/**
+		 * Runs actions immediately after the email address was successfully subscribed to the form.
+		 *
+		 * @since   1.2.1
+		 *
+		 * @param   array   $response   API Response
+		 * @param   int     $form_id    Form ID
+		 * @param   string  $email      Email Address
+		 * @param   string  $first_name First Name
+		 * @param   mixed   $fields     Custom Fields (false|array)
+		 * @param   mixed   $tag_ids    Tags (false|array)
+		 */
+		do_action( 'convertkit_api_form_subscribe_success', $response, $form_id, $email, $first_name, $fields, $tag_ids );
+
+		return $response;
+
+	}
+
+	/**
+	 * Gets all landing pages from the API.
+	 *
+	 * @since   1.9.6
+	 *
+	 * @return  WP_Error|array
+	 */
+	public function get_landing_pages() {
+
+		$this->log( 'API: get_landing_pages()' );
+
+		// Get all forms and landing pages from the API.
+		$forms = $this->get_forms_landing_pages();
+
+		// If an error occured, log and return it now.
+		if ( is_wp_error( $forms ) ) {
+			$this->log( 'API: get_landing_pages(): Error: ' . $forms->get_error_message() );
+			return $forms;
+		}
+
+		return $forms['landing_pages'];
+
+	}
+
+	/**
+	 * Fetches all sequences from the API.
+	 *
+	 * @since   1.9.6
+	 *
+	 * @return  WP_Error|array
+	 */
+	public function get_sequences() {
+
+		$this->log( 'API: get_sequences()' );
+
+		$sequences = array();
+
+		// Send request.
+		$response = $this->get(
+			'sequences',
+			array(
+				'api_key' => $this->api_key,
+			)
+		);
+
+		// If an error occured, log and return it now.
+		if ( is_wp_error( $response ) ) {
+			$this->log( 'API: get_sequences(): Error: ' . $response->get_error_message() );
+			return $response;
+		}
+
+		// If no sequences exist, return WP_Error.
+		if ( ! isset( $response['courses'] ) ) {
+			$this->log( 'API: get_sequences(): Error: No sequences exist in ConvertKit.' );
+			return new WP_Error( 'convertkit_api_error', $this->get_error_message( 'get_sequences_none' ) );
+		}
+		if ( ! count( $response['courses'] ) ) {
```

Comment:
Should this really throw an exception when there are no items? It seems like the consumer should make this choice, not the API class. This comment goes for all the other functions that throw an error for zero items.

Reply:
Sounds good. There's a separate PR for this change [here](https://github.com/ConvertKit/convertkit-wordpress/pull/330).